src/server/worker/driver/driver_unix.go

```diff
@@ -0,0 +1,156 @@
+// +build !windows
+
+package driver
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/server/worker/common"
+)
+
+// Mkfifo does not exist on Windows, so this is left unimplemented there, except for tests
+func createSpoutFifo(path string) error {
+	return syscall.Mkfifo(path, 0666)
+}
+
+func makeCmdCredentials(uid uint32, gid uint32) *syscall.SysProcAttr {
+	return &syscall.SysProcAttr{
+		Credential: &syscall.Credential{
+			Uid: uid,
+			Gid: gid,
+		},
+	}
+}
+
+// WithActiveData is implemented differently in unix vs windows because of how
+// symlinks work on windows. Here, we create symlinks to the scratch space
+// directory, then clean up before returning.
+func (d *driver) WithActiveData(inputs []*common.Input, dir string, cb func() error) (retErr error) {
+	d.activeDataMutex.Lock()
+	defer d.activeDataMutex.Unlock()
+
+	if err := d.linkData(inputs, dir); err != nil {
+		return errors.Wrap(err, "error when linking active data directory")
+	}
+	defer func() {
+		if !d.PipelineInfo().S3Out {
+			if err := d.rewriteSymlinks(dir); err != nil && retErr == nil {
+				retErr = errors.Wrap(err, "error when redirecting symlinks in the active data directory")
+			}
+		}
+		if err := d.unlinkData(inputs); err != nil && retErr == nil {
+			retErr = errors.Wrap(err, "error when unlinking active data directory")
+		}
+	}()
+
+	return cb()
+}
+
+// splitPath will split the given path into an array containing each directory
+// component followed by the filename
+func splitPath(path string) []string {
+	result := []string{}
+	for {
+		_, component := filepath.Split(path)
+		path = filepath.Dir(path)
+
+		if component == "" {
+			// reverse the result
+			for i, j := 0, len(result)-1; i < j; i, j = i+1, j-1 {
+				result[i], result[j] = result[j], result[i]
+			}
+			return result
+		}
+
+		result = append(result, component)
+	}
+}
```

Comment:
Does this give a different result than `strings.Split(filepath.Clean(path), "/")`? That's what I normally wind up using in cases like this, if it does then this should probably go in a util package somewhere, I think there's a lot of places where we do something similar to this.

Reply:
I think you're right, I just expected this to be a function in the `filepath` package and it wasn't.